src/expression.c

```diff
@@ -6104,8 +6104,9 @@ Expression *TypeidExp::semantic(Scope *sc)
     {
         /* Get the dynamic type, which is .classinfo
          */
-        e = new DotIdExp(ea->loc, ea, Id::classinfo);
-        e = e->semantic(sc);
+        ea = ea->semantic(sc);
+        e = new TypeidExp(ea->loc, ea);
+        e->type = Type::typeinfoclass->type;
```

Comment:
I thought `Type::vtinfo` _was_ the `TypeInfo_Class` instance.  What is it?

Wait, is `TypeInfo_Class` the typeinfo for the class reference, but classinfo is the typeinfo for the actual class instance?  I guess that would explain why they have the same name...